packages/builders/src/components/button/UnsafeButton.ts

```diff
@@ -126,4 +127,17 @@ export class UnsafeButtonComponent extends Component<Partial<APIButtonComponent>
 		}
 		return isEqual(other, this.data);
 	}
+
+	public build() {
+		Object.freeze(this.data);
+		return buildComponent<BuildButton>(this);
+	}
 }
+
+type BuildButton = Omit<
+	UnsafeButtonComponent,
+	'setStyle' | 'setURL' | 'setCustomId' | 'setEmoji' | 'setDisabled' | 'setLabel' | 'equals' | 'build' | 'url'
+> & {
+	customId: string;
+	style: Omit<ButtonStyle, 'LINK'>;
```

Comment:
Is there a reason `LINK` is omitted?

Reply:
You can't receive a link button interaction.